pkg/apis/workflow/v1alpha1/pod_template_types.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+type PodTemplate struct {
+	Graph        ContainerGraph       `json:"graph,omitempty" protobuf:"bytes,1,rep,name=graph"`
+	Sequence     ContainerSequence    `json:"sequence,omitempty" protobuf:"bytes,2,rep,name=sequence"`
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty" protobuf:"bytes,3,rep,name=volumeMounts"`
+}
+
+func (in *PodTemplate) GetContainers() []corev1.Container {
+	var ctrs []corev1.Container
+	for _, t := range in.GetGraph() {
+		c := t.Container
+		c.VolumeMounts = append(c.VolumeMounts, in.VolumeMounts...)
+		ctrs = append(ctrs, c)
+	}
+	return ctrs
+}
+
+func (in *PodTemplate) HasContainerNamed(n string) bool {
+	for _, c := range in.GetContainers() {
+		if n == c.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (in *PodTemplate) GetGraph() []ContainerNode {
+	if in == nil {
+		return nil
+	}
+	out := in.Graph.Containers
+	for i, t := range in.Sequence.Containers {
+		var d []string
+		if i > 0 {
+			d = append(d, in.Sequence.Containers[i-1].Name)
+		}
+		out = append(out, ContainerNode{Container: t, Dependencies: d})
+	}
+	return out
+}
+
+func (in *PodTemplate) HasSequencedContainers() bool {
+	for _, n := range in.GetGraph() {
+		if len(n.Dependencies) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type ContainerGraph struct {
+	Containers []ContainerNode `json:"containers,omitempty" protobuf:"bytes,1,rep,name=containers"`
+}
+
+type ContainerNode struct {
+	corev1.Container `json:",inline" protobuf:"bytes,1,opt,name=container"`
+	Dependencies     []string `json:"dependencies,omitempty" protobuf:"bytes,2,rep,name=dependencies"`
```

Comment:
I think we should either:

* Only support `depends` to match [enhanced depends](https://github.com/argoproj/argo-workflows/blob/master/docs/enhanced-depends-logic.md)
* Support both `depends` and `dependencies` as we currently do for DAG

Reply:
Interesting... that might not be too hard to do and a LOT more powerful.

Reply:
v1.1?

Reply:
I made my below comment before reading @simster7 comment but I agree with simon. I think we should support only one syntax, which is the more powerful`depends`. If it is too much to support `||` operation for the MVP, we can only deliver `&&` for v1.